common/oidc-client/oauth2-proxy/components/configure-self-signed-kubernetes-oidc-issuer/cronjob.kubeflow-m2m-oidc-configurator.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: kubeflow-m2m-oidc-configurator
+  namespace: istio-system
+spec:
+  schedule: '* * * * *'
+  concurrencyPolicy: Forbid
+  jobTemplate:
+    spec:
+      ttlSecondsAfterFinished: 60
+      template:
+        metadata:
+          labels: {}
+        spec:
+          restartPolicy: OnFailure
+          serviceAccountName: kubeflow-m2m-oidc-configurator
+          containers:
+            - image: curlimages/curl
+              name: kubeflow-m2m-oidc-configurator
+              command:
+                - /script.sh
+              envFrom:
+                - configMapRef:
+                    name: kubeflow-m2m-oidc-configurator-envs
+              volumeMounts:
+                - mountPath: /script.sh
+                  name: script
+                  subPath: script.sh
+              resources: {}
+          volumes:
+            - name: script
+              configMap:
+                name: kubeflow-m2m-oidc-configurator-script
+                defaultMode: 0777
+                items:
+                  - key: script.sh
+                    path: script.sh
```

Comment:
Huh, well it doesn't verify if the JWKS is present and after all is always performing the patch so this might be an improvement. I think the JWKS value should be also compared and only patched if different.

Reply:
I made changes so the script will first check for the JWKS present in RequestAuthentication and only patch if not equal to the desired JWKS.